### ISSUE TITLE
Error when updating Max Cache Entry Size

### ISSUE THREAD

**Issue by oakromulo** [metabase/metabase#9143]
**Browser**: Chrome: 70.0.3538.110 (Official Build) (64-bit)
**OS**: macOS Mojave 10.14.1 (18B75)
**Database**: AWS Redshift 1.0.5290 (irrelevant to the issue)
**Metabase version**: 0.31.2
**Metabase hosting**: Docker / AWS EC2 / Amazon Linux 2018.3 (centOS)
**Metabase internal database**: Postgres 10

Steps to **reproduce**:

Under:
Metabase Admin -> Caching -> Input any value on MAX CACHE ENTRY SIZE with Caching Enabled

Error: java.lang.String cannot be cast to java.lang.Number

![screen shot 2018-12-17 at 2 43 45 pm](https://user-images.githubusercontent.com/19331468/50101643-490c9280-020a-11e9-9603-bbd3fe4d66fd.png)

**Comment by mazameli** [comment]
Reproduced this. Here are the server logs I'm seeing:

```
Dec 17 09:42:26 ERROR metabase.middleware :: PUT /api/setting/query-caching-max-kb 500 (13 ms) (0 DB calls).
{:message "java.lang.String cannot be cast to java.lang.Number",
:type java.lang.ClassCastException,
:stacktrace
("clojure.lang.Numbers.gt(Numbers.java:234)"
"clojure.lang.Numbers.gt(Numbers.java:3889)"
"--> public_settings$eval35512$fn__35513.invoke(public_settings.clj:111)"
"models.setting$set_BANG_.invokeStatic(setting.clj:452)"
"models.setting$set_BANG_.invoke(setting.clj:440)"
"api.setting$fn__63730.invokeStatic(setting.clj:37)"
"api.setting$fn__63730.invoke(setting.clj:31)"
"middleware$enforce_authentication$fn__53082.invoke(middleware.clj:113)"
"api.routes$fn__64307.invokeStatic(routes.clj:82)"
"api.routes$fn__64307.invoke(routes.clj:82)"
"routes$fn__64358$fn__64359.doInvoke(routes.clj:108)"
"routes$fn__64358.invokeStatic(routes.clj:103)"
"routes$fn__64358.invoke(routes.clj:103)"
"middleware$catch_api_exceptions$fn__53231.invoke(middleware.clj:436)"
"middleware$log_api_call$fn__53209$fn__53211.invoke(middleware.clj:364)"
"middleware$log_api_call$fn__53209.invoke(middleware.clj:363)"
"middleware$add_security_headers$fn__53137.invoke(middleware.clj:252)"
"core$wrap_streamed_json_response$fn__65419.invoke(core.clj:67)"
"middleware$bind_current_user$fn__53087.invoke(middleware.clj:137)"
"middleware$maybe_set_site_url$fn__53147.invoke(middleware.clj:290)"
"middleware$add_content_type$fn__53140.invoke(middleware.clj:262)")}

Dec 17 09:42:50 ERROR metabase.driver.sql-jdbc.execute :: nil
Dec 17 09:42:50 ERROR metabase.driver.sql-jdbc.execute
```

**Comment by mralexpopa** [comment]
Came here to report the same issue. Doesn't seem possible to change the cache size to any other value.

**Comment by flamber** [comment]
@mralexpopa Which version are you using? This is fixed in the upcoming 0.32.0 release.
